department-of-veterans-affairs/caseflow
# seed data + UI for scenario 1

Resolves [APPEALS-61641](https://jira.devops.va.gov/browse/APPEALS-61641)

# Description
Created a new Task Action for Legacy Appeals With an Active Blocking Hearing Task, when the user is a SCM User, that allows them to cancel all blocking tasks and advance the case to a judge.

Connected Task Action to expected UI Modals and routes.

Added seed data to the db/seeds/legacy_appeals_for_vlj_movement_testing.rb file to test my new scenario.

Fixed bug in Case Details page where Legacy Appeals with Available Task Actions on an Active Schedule Hearing Task would throw an error when selecting the button View All Cases. Issue was due to missing attributes in the Legacy Appeal Search Serializer. This bug was found in demo but affected all environments including Prod. 

## Acceptance Criteria
1. When the SCM group member is on the Case Details page of the selected undistributed legacy appeal that currently has an open blocking hearing task, they can select Case Movement: Cancel blocking tasks and advance to judge from the actions menu associated for the hearing task. 
2. The following appeals have been created to support testing of this functionality:

Number of appeals | Veteran FN | Veteran LN | Pri/not | #issues | blocking hearing task | notes
-- | -- | -- | -- | -- | -- | --
2 | HearBlock | One | y, age | 1 | Schedule hearing |  
2 | HearBlock | Two | n | 2 |   |  
2 | HearBlock | Three | y, manual | 2 | Other blocking |  
1 | HearBlock | CAVC | n | 1 | Schedule hearing | this should be a CAVC remand
1 | HearBlock | Four | n | 1 |   |  
1 | HearNon | Block | Y | 1 | n | has non-blocking task
1 | DoesNot | Qualify | N | 1 |   | has location of 83


## Testing Plan

1. Run seed data using the command `bundle exec rake db:seed:legacy_appeals_for_vlj_movement_testing` && enable the feature toggle `FeatureToggle.enable!(:legacy_case_movement_scm_to_vlj_for_blockhtask)`
2. Switch to User "TEAM_ADMIN" 
3. Go to their queue, switch views to the Board Of Veteran's Appeals Cases
4. Filter by Legacy Appeal to find the specific seeds with their Veteran Name
5. Open all appeals, then verify no task action is available on the active blocking hearing task.
6. With the appeals open, in another tab switch users to "BVARDUNKLE" who is an SCM User.
7. Refresh the appeal case details pages
8. Verify that there is a Task Action dropdown next to the HearingTask that allows the SCM User to Cancel Blocking Tasks and Advance To Judge

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
